sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -212,31 +212,96 @@ class Analyzer(
   )
 
   /**
-   * Analyze cte definitions and substitute child plan with analyzed cte definitions.
+   * Analyze WITH nodes and substitute child plan with CTE definitions.
    */
   object CTESubstitution extends Rule[LogicalPlan] {
```

Comment:
How about making a new file for this rule? cc: @gatorsmile

Reply:
ok moved